services/catalog/service/src/domains/channels/handlers/channel-published-event-handler.db.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { insert, select, selectOne } from 'zapatos/db';
+import { channel } from 'zapatos/schema';
+import {
+  createChannelPublishedEvent,
+  createTestContext,
+  ITestContext,
+} from '../../../tests/test-utils';
+import { ChannelPublishedEventHandler } from './channel-published-event-handler';
+
+describe('ChannelPublishEventHandler', () => {
+  let ctx: ITestContext;
+  let handler: ChannelPublishedEventHandler;
+
+  beforeAll(async () => {
+    ctx = await createTestContext();
+    handler = new ChannelPublishedEventHandler(ctx.loginPool, ctx.config);
+  });
+
+  afterEach(async () => {
+    await ctx?.truncate('channel');
+  });
+
+  afterAll(async () => {
+    await ctx?.dispose();
+    jest.restoreAllMocks();
+  });
+
+  describe('onMessage', () => {
+    test('A new channel is published', async () => {
+      // Arrange
+      const message = createChannelPublishedEvent('channel-1');
```

Comment:
Managed entities use UUIDs. So I guess it would be best to reflect that also in our tests.

Reply:
Adjusted tests to use UUIDs for Channel unique identifier.